### PR TITLE
Parameterize the build image docker hub repo name

### PR DESCRIPTION
**Pull request by jose-bigio**
Parameterize the build image docker hub repo name

Signed-off-by: jose-bigio <jose.bigio@docker.com>

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by tmjd** [comment]
LGTM

**Comment by fasaxc** [comment]
@jose-bigio Looks like you've got a conflict, please an you fix up?

File: Makefile [hunk 1]
```diff
@@ -49,15 +49,16 @@ CNI_VERSION=v0.6.0
...
 
 # By default set the CNI_SPEC_VERSION to 0.3.1 for tests.
 CNI_SPEC_VERSION?=0.3.1
 
 DIST=dist/$(ARCH)
 # Ensure that the dist directory is always created
 MAKE_SURE_DIST_EXIST := $(shell mkdir -p $(DIST))
-CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)
-CONTAINER_NAME=calico/cni
+CALICO_BUILD?=$(BUILD_IMAGE_ORG)/go-build$(ARCHTAG):$(GO_BUILD_VER)
```
Comment (fasaxc) [inline Makefile]: This line and the one below look incorrect for `master`.  On `master`, the arch is part of the image tag.  For example: `calico/go-build:v0.15-amd64`

In the release-v3.x branches we hadn't settled on that naming scheme yet.

File: Makefile [hunk 2]
```diff
@@ -52,18 +52,19 @@ CNI_VERSION=v0.6.0
...
 BIN=bin/$(ARCH)
 # Ensure that the bin directory is always created
 MAKE_SURE_BIN_EXIST := $(shell mkdir -p $(BIN))
-CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)
+CALICO_BUILD?=$(BUILD_IMAGE_ORG)/go-build:$(GO_BUILD_VER)$(ARCHTAG)
 
 PACKAGE_NAME?=github.com/projectcalico/cni-plugin
 
-CONTAINER_NAME=calico/cni
+CONTAINER_NAME=calico/cni$(ARCHTAG)
```
Comment (fasaxc) [inline Makefile]: This makefile generally uses `calico/cni:latest-<arch>` so I'm not sure why we need this change?

File: Makefile [hunk 3]
```diff
@@ -52,18 +52,19 @@ CNI_VERSION=v0.6.0
...
+BUILD_IMAGE_ORG?=calico
 
 # By default set the CNI_SPEC_VERSION to 0.3.1 for tests.
 CNI_SPEC_VERSION?=0.3.1
 
 BIN=bin/$(ARCH)
 # Ensure that the bin directory is always created
 MAKE_SURE_BIN_EXIST := $(shell mkdir -p $(BIN))
-CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)
+CALICO_BUILD?=$(BUILD_IMAGE_ORG)/go-build:$(GO_BUILD_VER)$(ARCHTAG)
```
Comment (deitch) [inline Makefile]: Same as @fasaxc 's comment on L67. `ARCHTAG` isn't set at all, and `calico/go-build` is a multi-arch manifest anyways. Why use it here?

File: Makefile (hunk 2 above)
Comment (jose-bigio) [reply Makefile]: That's an oversight on my part then.

**Pull request merged**